torch/library.h

```diff
@@ -435,8 +478,8 @@ class CAFFE2_API Library final {
   ///   m.impl("add", add_cuda);
   /// }
   /// ```
-  template <typename Func>
-  Library& impl(const char* name, Func&& raw_f) & {
+  template <typename Name, typename Func>
```

Comment:
This pattern opens up `impl` to be called with any type for `Name` and will yield very bad error messages when called with non-string types or string types that we don't support. Can we keep the old implementation (or maybe actually change it to take string_view instead of const char*) and then explicitly specialize for `SelectiveStr<true>`? The specialization could just forward to the `string_view` version.